tools/extract_features.cpp

```diff
@@ -158,10 +160,13 @@ int feature_extraction_pipeline(int argc, char** argv) {
         }
         int length = snprintf(key_str, kMaxKeyStrLength, "%d",
             image_indices[i]);
-        CHECK(feature_dbs.at(i)->put(std::string(key_str, length), datum));
+        string out;
+        CHECK(datum.SerializeToString(&out));
+        txns.at(i)->Put(std::string(key_str, length), out);
         ++image_indices[i];
         if (image_indices[i] % 1000 == 0) {
-          CHECK(feature_dbs.at(i)->commit());
+          txns.at(i)->Commit();
+          txns.at(i).reset(feature_dbs.at(i)->NewTransaction());
```

Comment:
See http://symas.com/mdb/doc/group__mdb.html#ga846fbd6f46105617ac9f4d76476f6597; transactions are freed after commit by LMDB, there is no other way (and they should be freed properly if not committed after the next fix).